### PR TITLE
Informs MAC users that docker is required on make shell

### PR DESCRIPTION
**Pull request by rsertelon**
This should address #1397

**Warning**: I don't have a MAC, I just tested with `IN_DOCKER=true make shell` to see if my error actually worked. But under linux...

Someone with a mac should test it :)

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: Makefile [hunk 1]
```diff
@@ -156,6 +156,9 @@ serve-docs: docs ## serves the project documentation from an HTTP server
 .PHONY: serve-docs
 
 ifeq ($(IN_DOCKER),true)
+ifeq (, $(shell which docker))
+	$(error "Docker does not seem installed, please install docker first.")
+endif
```
Comment (christophermaier) [inline Makefile]: This doesn't work for me on my MacOS workstation.

I think you might need to do something like:

```
...
HAS_DOCKER := $(shell command -v docker 2> /dev/null)
...

image:
    ifndef HAS_DOCKER
        $(error "Docker does not seem to be installed....")
    endif
    # other stuff

```

Comment (rsertelon) [reply Makefile]: What is your error exactly?

Comment (christophermaier) [reply Makefile]: It was the same error as reported in the original issue (#1397)

**Comment by christophermaier** [comment]
tenor-99593459
Works on my machine 😄 

@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/296451023).

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/296451023).

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/296451023).

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/296451023).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/296451023).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Comment by martinisoft** [comment]
Thank you! ❤️

**Pull request merged**